packages/backend/src/nest/registration/registration.service.ts

```diff
@@ -18,32 +22,15 @@ export class RegistrationService extends EventEmitter implements OnModuleInit {
   onModuleInit() {
     this.on(
       RegistrationEvents.REGISTER_USER_CERTIFICATE,
-      async (payload: { csrs: string[]; certificates: string[]; id: string }) => {
-        await this.issueCertificates(payload)
+      (payload: { csrs: string[]; certificates: string[]; id: string }) => {
+        this.registrationEvents.push({ csrs: payload.csrs, id: payload.id })
+        this.tryRegisterNextUserCertificates()
```

Comment:
tryRegisterNextUserCertificates() is async function so it needs to be `await`ed. Otherwise it's a fire-and-forget, which is a very bad idea

I think something may be broken again with our eslint. Do you not get an error on this line? It should complain about missing await. If you *did* want to actually fire-and-forget you should write it as 
```
void this.tryRegisterNextUserCertificates()
```
but this is asually a very bad idea and rarely the behaviour you want

Reply:
I think

```
new Promise(something).then(another),
```

is equivalent to 

```
test = async () => {
  await something
  another
}
```

and where the Promise body, `something` is entirely synchronous. I think it's only the promise callback, `another`, which is added to the event loop and is asynchronous. So if there is no line after an `await`, I don't think the `await` does anything. I don't think the `async` keyword actually makes the function async. It's still synchronous unless there is a promise callback or I/O callback or timer. I've been reading a bit more about how JS works internally and posted a lot of links in the Slack internal channel. I also wrote some interested tests over here: https://github.com/TryQuiet/quiet/compare/develop...race

I was confused myself and so I started diving into it a bit more. I don't have the whole picture and am likely still confused about many things, but this is my latest understanding

Reply:
> So if there is no line after an await, I don't think the await does anything. I don't think the async keyword actually makes the function async

It's actually the opposite.

If you don't `await` this means that that the outer Promise will fire immediately. If you `await` the inner Promise the outer promise will wait for the inner one.
Even more importantly, if inner Promise throws a failure:
a) with await the outer Promise will be a failuer
b) without await the outer Promise will succeed and the failure from inner Promise will hit the generic error handler of the process

This second point is the main reason for always awaiting stuff. Without it, error handling is a nightmare.